### PR TITLE
Fixees net UI lists AAAAAA

### PR DESCRIPTION
**Pull request by Bod9001**
### Changelog:

CL: [Improvement] banana floor tiles actually are worth something 250 credits

CL: [Balance] removed required access for Nonsecurity related vending machines

CL: [Fix] Dynamic entries in UIs like prolathe and autolathe have been fixed.

### CODE REVIEW

**Comment by MaxIsJoe** [comment]
`CL: [Fix] Dynamic entries in UIs like prolathe and autolathe have been fixed.`

File: UnityProject/Assets/Scripts/UI/Core/Net/Elements/Dynamic/NetUIDynamicList.cs [hunk 1]
```diff
@@ -106,11 +91,14 @@ public override string ToString()
 		{
 			return string.Join(",", Value);
 		}
-
 		public virtual void Clear()
 		{
-			DisabledEntryPool.EnqueueAll(Entries.ToList());
-
+			foreach (var Entry in Entries)
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/UI/Core/Net/Elements/Dynamic/NetUIDynamicList.cs]:
```suggestion
			foreach (var entry in Entries)
```

File: UnityProject/Assets/Scripts/UI/Core/Net/Elements/Dynamic/NetUIDynamicList.cs [hunk 2]
```diff
@@ -106,11 +91,14 @@ public override string ToString()
...
+			{
+				DestroyImmediate(Entry.gameObject);
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/UI/Core/Net/Elements/Dynamic/NetUIDynamicList.cs]:
```suggestion
				DestroyImmediate(entry.gameObject);
```

File: UnityProject/Assets/Scripts/UI/Core/Net/Elements/Dynamic/NetUIDynamicList.cs [hunk 3]
```diff
@@ -192,17 +181,10 @@ protected DynamicEntry[] AddBulk(string[] proposedIndices)
 
 		private DynamicEntry PoolSpawnEntry()
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/UI/Core/Net/Elements/Dynamic/NetUIDynamicList.cs]: the new body for this method looks like this isn't pooled anymore. Is that intended?

File: UnityProject/Assets/Scripts/UI/Core/Net/Elements/NetButton.cs [hunk 4]
```diff
@@ -23,7 +23,11 @@ public override void ExecuteServer(PlayerInfo subject)
 		public void Awake()
 		{
 			Button = this.GetComponent<Button>();
+			Button.onClick = new Button.ButtonClickedEvent();
 			Button.onClick.AddListener(ExecuteClient);
 		}
+
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/UI/Core/Net/Elements/NetButton.cs:29]: what's up with the new empty lines?

**Review by corp-0 — CHANGES_REQUESTED** [review]
changes in individual comments

File: UnityProject/Assets/Scripts/UI/Core/Net/Elements/NetButton.cs (hunk 4 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/UI/Core/Net/Elements/NetButton.cs:29]: was debugging stuff and didn't properly cleanup the old functions

File: UnityProject/Assets/Prefabs/Items/Construction/PlaceableTiles/MineralTiles/BananiumTile.prefab [hunk 5]
```diff
@@ -45,6 +45,11 @@ PrefabInstance:
       value: 
       objectReference: {fileID: 11400000, guid: 83855008e43a2674d8d6a006ce94326f,
         type: 2}
+    - target: {fileID: 6610653795340409978, guid: b122d05362087574ca3b106604026dd9,
+        type: 3}
+      propertyPath: exportCost
+      value: 250
```
Comment (corp-0) [inline UnityProject/Assets/Prefabs/Items/Construction/PlaceableTiles/MineralTiles/BananiumTile.prefab:51]: this change looks to be unrelated to the PR. At least register the change in the changelog for the players to know it was tweaked.

File: UnityProject/Assets/Scripts/UI/Core/Net/Elements/Dynamic/NetUIDynamicList.cs (hunk 3 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/UI/Core/Net/Elements/Dynamic/NetUIDynamicList.cs]: yeah, pooling had weird behaviour

File: UnityProject/Assets/Prefabs/Items/Construction/PlaceableTiles/MineralTiles/BananiumTile.prefab (hunk 5 above)
Comment (Bod9001) [reply UnityProject/Assets/Prefabs/Items/Construction/PlaceableTiles/MineralTiles/BananiumTile.prefab:51]: "CL: [Improvement] banana floor tiles actually are worth something 250 credits"

File: UnityProject/Assets/Scripts/UI/Core/Net/Elements/Dynamic/NetUIDynamicList.cs (hunk 3 above)
Comment (corp-0) [reply UnityProject/Assets/Scripts/UI/Core/Net/Elements/Dynamic/NetUIDynamicList.cs]: in that case, the method has to be renamed
